### PR TITLE
Feature/overlapping reads

### PR DESCRIPTION
**Pull request by keiranmraine**
@drjsanger Could you review the stale branches too as there's an old hotfix in there.  I'd like to get a new version (v4.0.0 as we've changes a fair bit) out so that I can reference it in the ascatNgs release.

### CODE REVIEW

File: c/src/alleleCounter.c [hunk 1]
```diff
@@ -86,10 +92,11 @@ void alleleCounter_setup_options(int argc, char *argv[]){
              	{"min-base-qual", required_argument, 0, 'm'},
 							{"min-map-qual", required_argument, 0, 'q'},
 							{"is-snp6", required_argument, 0, 's'},
+							{"is-10x", required_argument, 0, 'x'},
 							{"contig", required_argument, 0, 'c'},
 							{"dense-snps", no_argument, 0, 'd'},
-							{"required-flag", no_argument, 0, 'f'},
-							{"filtered-flag", no_argument, 0, 'F'},
+							{"required-flag", optional_argument, 0, 'f'},
+							{"filtered-flag", optional_argument, 0, 'F'},
```
Comment (drjsanger) [inline c/src/alleleCounter.c]: `optional_argument` should be `required_argument` . The naming is misleading. But optional_argument actually means that the argument can optionally have a value. These both require a value to be given if the argument is set [docced here](https://linux.die.net/man/3/getopt_long)

File: CHANGES.md [hunk 2]
```diff
@@ -1,24 +1,39 @@
-### v3.3.1
+# CHANGES
+
+## NEXT
+
+* alleleCounter now counts per-fragment rather than per-read when overlaps occur.
+* Reworked perl to wrap C alleleCounter and just handle the extra format changes.
+* No dep on Bio::DB::HTS now.
+* Update to HTSlib 1.7
+
```
Comment (drjsanger) [inline CHANGES.md]: 10X code added?

Comment (drjsanger) [reply CHANGES.md]: Have cleaned up other branches

File: c/src/alleleCounter.c (hunk 1 above)
Comment (keiranmraine) [reply c/src/alleleCounter.c]: @drjsanger Ready for re-review

File: CHANGES.md (hunk 2 above)
Comment (keiranmraine) [reply CHANGES.md]: @drjsanger Ready for re-review

**Pull request merged**